[Bug] Karpenter fails to provision

<!--
Thank you for opening this bug report! You are helping make eksctl a better project :)
In order to help us process this issue faster, please provide as much detail as possible.

When providing command examples, error output or code samples, please use code blocks.
For verbose output, for example logs, please either create a gist or use html details tags.
-->

<!--
Before you start writing:
1. Please search through existing issues to see whether your problem has already been reported.
2. Download the latest version of eksctl to see whether the problem has already been fixed.

For more information on how to open a bug report in eksctl, please refer to our issue submission guide.
https://github.com/weaveworks/eksctl/blob/main/CONTRIBUTING.md#opening-issues
-->

### What were you trying to accomplish?
<!-- Information about the command you ran and what you expected to happen. -->
Created a cluster with the following configuration

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: karp
  region: us-east-2
  version: '1.24'
  tags:
    karpenter.sh/discovery: karp

iam:
  withOIDC: true

karpenter:
  version: 'v0.20.0'
  createServiceAccount: true

managedNodeGroups:
  - name: addons-ng-1
    minSize: 2
    maxSize: 5
    desiredCapacity: 2
    taints:
      - key: CriticalAddonsOnly
        value: "true"
        effect: NoSchedule

addons:
  - name: coredns
  - name: aws-ebs-csi-driver
    wellKnownPolicies:
      ebsCSIController: true
  - name: vpc-cni
```

### What happened?
<!-- A description of actual behavior (with error messages). -->
The cluster created with karpenter installed but pods were crashing because they were missing settings. The karpenter-global-settings configmap in the karpenter namespace was missing the settings for 

```
   aws.clusterEndpoint: ""
   aws.clusterName: ""
   aws.defaultInstanceProfile: ""
```
After filling in this information the controller would run but was giving the following error in the logs
```
ERROR   webhook.ValidationWebhook       Reconcile error {"commit": "f60dacd", "knative.dev/traceid": "5680f5d4-871
b-46e2-a32e-9808d5b88fdb", "knative.dev/key": "karpenter/karpenter-cert", "duration": "115.870254ms", "error": "failed to update webhook: Operatio
n cannot be fulfilled on validatingwebhookconfigurations.admissionregistration.k8s.io \"validation.webhook.karpenter.k8s.aws\": the object has bee
n modified; please apply your changes to the latest version and try again"}
```
I had to apply the latest karpenter CRDs to the cluster with
```
kubectl replace -f https://raw.githubusercontent.com/aws/karpenter/v0.20.0/pkg/apis/crds/karpenter.sh_provisioners.yaml
kubectl replace -f https://raw.githubusercontent.com/aws/karpenter/v0.20.0/pkg/apis/crds/karpenter.k8s.aws_awsnodetemplates.yaml
```
At this point I could create the karpenter provisioner and AWSNodeTemplate and then I was ready to 

### How to reproduce it?
<!--
Include ALL the steps to reproduce the bug.

If using a config file, include it here, removing any sensitive information!
-->
Use the config provided with `eksctl create cluster -f config.yaml`

**Logs**
<!--
Include the output of the command line when running eksctl. If possible, eksctl should be run with debug logs. For example:
`eksctl get clusters -v 4`
Make sure you redact any sensitive information before posting.
If the output is long, please consider a Gist, or an html details tag.
-->

**Anything else we need to know?**
<!--
What OS are you using?
Are you using a downloaded binary or did you compile eksctl?
What type of AWS credentials are you using (i.e. default/named profile, MFA)? - please don't include actual credentials though!
-->

**Versions**
<!--
Please paste in the output of this command:
-->
```
$ eksctl info
eksctl version: 0.122.0-dev+b7686d461.2022-11-29T18:34:06Z
kubectl version: v1.25.4
OS: linux
```

I've got the same issue while using eks `v1.24` and karpenter `v0.20.0`. Here my eksctl and kubectl versions:
```
eksctl version: 0.124.0
kubectl version: v1.24.7-eks-fb459a0
OS: linux
```

Here are the logs from the carpenter controller:

```
kubectl logs -n karpenter karpenter-645754498f-jx7sp controller --follow
2022-12-28T17:50:18.417Z        DEBUG   Successfully created the logger.
2022-12-28T17:50:18.417Z        DEBUG   Logging level set to: debug
{"level":"info","ts":1672249818.421702,"logger":"fallback","caller":"injection/injection.go:63","msg":"Starting informers..."}
2022-12-28T17:50:18.522Z        DEBUG   controller      waiting for configmaps  {"commit": "f60dacd", "configmaps": ["karpenter-global-settings"]}
E1228 17:50:19.026538       1 runtime.go:77] Observed a panic: validating settings, "" not a valid clusterEndpoint URL; Key: 'Settings.ClusterName' Error:Field validation for 'ClusterName' failed on the 'required' tag
Key: 'Settings.ClusterEndpoint' Error:Field validation for 'ClusterEndpoint' failed on the 'required' tag
goroutine 126 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x213fae0?, 0xc000367c90})
        k8s.io/apimachinery@v0.25.4/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0009831f0?})
        k8s.io/apimachinery@v0.25.4/pkg/util/runtime/runtime.go:49 +0x75
panic({0x213fae0, 0xc000367c90})
        runtime/panic.go:884 +0x212
github.com/aws/karpenter/pkg/apis/config/settings.NewSettingsFromConfigMap(0xc000825200)
        github.com/aws/karpenter/pkg/apis/config/settings/settings.go:93 +0x4fb
reflect.Value.call({0x228dac0?, 0x2865dc8?, 0x0?}, {0x274531b, 0x4}, {0xc000989c40, 0x1, 0x0?})
        reflect/value.go:584 +0x8c5
reflect.Value.Call({0x228dac0?, 0x2865dc8?, 0xc000588fa0?}, {0xc000989c40?, 0x203000?, 0x203000?})
        reflect/value.go:368 +0xbc
knative.dev/pkg/configmap.(*UntypedStore).OnConfigChanged(0xc0003fdcc0, 0xc000825200)
        knative.dev/pkg@v0.0.0-20221123154742-05b694ec4d3a/configmap/store.go:141 +0x185
knative.dev/pkg/configmap.(*ManualWatcher).OnChange(0xc0004d19b0, 0xc000825200)
        knative.dev/pkg@v0.0.0-20221123154742-05b694ec4d3a/configmap/manual_watcher.go:72 +0x106
knative.dev/pkg/configmap/informer.(*InformedWatcher).addConfigMapEvent(0x740000000008?, {0x270b2c0?, 0xc000825200?})
        knative.dev/pkg@v0.0.0-20221123154742-05b694ec4d3a/configmap/informer/informed_watcher.go:220 +0x38
knative.dev/pkg/configmap/informer.(*syncedCallback).Call(0xc00087a700, {0x270b2c0?, 0xc000825200?}, {0xc000588fa0, 0x19})
        knative.dev/pkg@v0.0.0-20221123154742-05b694ec4d3a/configmap/informer/synced_callback.go:94 +0x3d
knative.dev/pkg/configmap/informer.(*InformedWatcher).Start.func1({0x270b2c0?, 0xc000825200?})
        knative.dev/pkg@v0.0.0-20221123154742-05b694ec4d3a/configmap/informer/informed_watcher.go:158 +0x45
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        k8s.io/client-go@v0.25.4/tools/cache/controller.go:232
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        k8s.io/client-go@v0.25.4/tools/cache/shared_informer.go:818 +0x134
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x406d01?)
        k8s.io/apimachinery@v0.25.4/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000071f38?, {0x2c55f40, 0xc0008ecf00}, 0x1, 0xc000b361e0)
        k8s.io/apimachinery@v0.25.4/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x7b7ed0?, 0x3b9aca00, 0x0, 0x26?, 0xc000071f88?)
        k8s.io/apimachinery@v0.25.4/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
        k8s.io/apimachinery@v0.25.4/pkg/util/wait/wait.go:92
k8s.io/client-go/tools/cache.(*processorListener).run(0xc00017e800?)
        k8s.io/client-go@v0.25.4/tools/cache/shared_informer.go:812 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
        k8s.io/apimachinery@v0.25.4/pkg/util/wait/wait.go:75 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        k8s.io/apimachinery@v0.25.4/pkg/util/wait/wait.go:73 +0x85
panic: validating settings, "" not a valid clusterEndpoint URL; Key: 'Settings.ClusterName' Error:Field validation for 'ClusterName' failed on the 'required' tag
Key: 'Settings.ClusterEndpoint' Error:Field validation for 'ClusterEndpoint' failed on the 'required' tag [recovered]
        panic: validating settings, "" not a valid clusterEndpoint URL; Key: 'Settings.ClusterName' Error:Field validation for 'ClusterName' failed on the 'required' tag
Key: 'Settings.ClusterEndpoint' Error:Field validation for 'ClusterEndpoint' failed on the 'required' tag

goroutine 126 [running]:
```
I'll test it using a lower version of karpenter and post the results.

--- a/pkg/karpenter/karpenter.go
+++ b/pkg/karpenter/karpenter.go
@@ -28,6 +28,7 @@ const (
 	serviceAccountAnnotation = "annotations"
 	serviceAccountName       = "name"
 	settings                 = "settings"
+	interruptionQueueName    = "interruptionQueueName"
 )
 
 // Options contains values which Karpenter uses to configure the installation.
@@ -68,6 +69,7 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
 		},
 		serviceAccountName: DefaultServiceAccountName,
 	}
+
 	values := map[string]interface{}{
 		clusterName:     k.ClusterConfig.Metadata.Name,
 		clusterEndpoint: k.ClusterConfig.Status.Endpoint,
@@ -79,6 +81,7 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
 				defaultInstanceProfile: instanceProfileName,
 				clusterName:            k.ClusterConfig.Metadata.Name,
 				clusterEndpoint:        k.ClusterConfig.Status.Endpoint,
+				interruptionQueueName:  k.ClusterConfig.Metadata.Name,
 			},
 		},
 		serviceAccount: serviceAccountMap,

--- a/pkg/karpenter/karpenter.go
+++ b/pkg/karpenter/karpenter.go
@@ -27,6 +27,7 @@ const (
 	serviceAccount           = "serviceAccount"
 	serviceAccountAnnotation = "annotations"
 	serviceAccountName       = "name"
+	settings                 = "settings"
 )
 
 // Options contains values which Karpenter uses to configure the installation.
@@ -72,6 +73,13 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
 		clusterEndpoint: k.ClusterConfig.Status.Endpoint,
 		aws: map[string]interface{}{
 			defaultInstanceProfile: instanceProfileName,
+		},
+		settings: map[string]interface{}{
+			aws: map[string]interface{}{
+				defaultInstanceProfile: instanceProfileName,
+				clusterName:            k.ClusterConfig.Metadata.Name,
+				clusterEndpoint:        k.ClusterConfig.Status.Endpoint,
+			},
 		},
 		serviceAccount: serviceAccountMap,
 	}

--- a/pkg/karpenter/karpenter_test.go
+++ b/pkg/karpenter/karpenter_test.go
@@ -61,6 +61,13 @@ var _ = Describe("Install", func() {
 				aws: map[string]interface{}{
 					defaultInstanceProfile: "role/profile",
 				},
+				settings: map[string]interface{}{
+					aws: map[string]interface{}{
+						defaultInstanceProfile: "role/profile",
+						clusterName:            cfg.Metadata.Name,
+						clusterEndpoint:        cfg.Status.Endpoint,
+					},
+				},
 			}
 			Expect(args).To(Equal(providers.InstallChartOpts{
 				ChartName:       "oci://public.ecr.aws/karpenter/karpenter",
@@ -102,6 +109,13 @@ var _ = Describe("Install", func() {
 					},
 					aws: map[string]interface{}{
 						defaultInstanceProfile: "role/profile",
+					},
+					settings: map[string]interface{}{
+						aws: map[string]interface{}{
+							defaultInstanceProfile: "role/profile",
+							clusterName:            cfg.Metadata.Name,
+							clusterEndpoint:        cfg.Status.Endpoint,
+						},
 					},
 				}
 				Expect(opts.Values).To(Equal(values))

--- a/pkg/karpenter/karpenter_test.go
+++ b/pkg/karpenter/karpenter_test.go
@@ -66,6 +66,7 @@ var _ = Describe("Install", func() {
 						defaultInstanceProfile: "role/profile",
 						clusterName:            cfg.Metadata.Name,
 						clusterEndpoint:        cfg.Status.Endpoint,
+						interruptionQueueName:  cfg.Metadata.Name,
 					},
 				},
 			}
@@ -115,6 +116,7 @@ var _ = Describe("Install", func() {
 							defaultInstanceProfile: "role/profile",
 							clusterName:            cfg.Metadata.Name,
 							clusterEndpoint:        cfg.Status.Endpoint,
+							interruptionQueueName:  cfg.Metadata.Name,
 						},
 					},
 				}

